pulsar-client-cpp/lib/Commands.cc

```diff
@@ -43,16 +43,18 @@ SharedBuffer Commands::writeMessageWithSize(const BaseCommand& cmd) {
     return buffer;
 }
 
-SharedBuffer Commands::newPartitionMetadataRequest(BaseCommand& cmd, const std::string& topic, uint64_t requestId) {
+SharedBuffer Commands::newPartitionMetadataRequest(const std::string& topic, uint64_t requestId) {
+    BaseCommand cmd;
```

Comment:
To avoid creating a new BaseCommand (and all internal objects) for each lookup: 

```cpp
static BaseCommand cmd;
static boost::mutex mutex:

mutex.lock();
// rest of the method
mutex.unlock();
return buffer;
```